python/langsmith/client.py

```diff
@@ -3138,11 +3138,11 @@ def _evaluate_strings(self, prediction, reference=None, input=None, **kwargs) ->
 
 
 def _tracing_thread_drain_queue(
-    tracing_queue: Queue, limit: Optional[int] = None
+    tracing_queue: Queue, limit: int = 100, block: bool = True
 ) -> List[TracingQueueItem]:
     next_batch: List[TracingQueueItem] = []
     try:
-        while item := tracing_queue.get(block=True, timeout=0.25):
+        while item := tracing_queue.get(block=block, timeout=0.25):
```

Comment:
This logic is pretty confusing, summarizing for my own understanding:

1. We check the queue to see if anything is there. **if not**: wait 250 ms more and if nothing is there, return all the accumulated items. **if so**: move to step to
2. We pop an item off the queue and add it to the batch
3. If we hit the limit on batch size, return

Reply:
Is there a degenerate case where you only get a run added every 200ms that could lead to something like 200ms * 100 = 20s run delays?

How can we make this debuggable?

Reply:
I can make it so only the first call blocks

Reply:
There is a reason why I made it like this, if we special case the first item as you suggest, then it ends up creating a number of very small batches whenever requests into the queue are ramping up

Reply:
Pushed a commit which is a compromise

Reply:
FYI after some testing this compromise loses a lot of the effectivness of the batching behavior, ie ends up creating too many small batches, I think a delay is needed

Reply:
makes sense to me